### ISSUE TITLE
Allow searching for songs using artist's name.

### ISSUE THREAD

**Issue by SnipeXandrej** [jmshrv/finamp#471]
Right now it's only possible to search for songs (in the SONGS tab) by using the song's name.

I think it would really improve the user experience by also allowing to search for songs by using the Artist's name while in the "SONGS" tab.

Because right now if I want to find a song named "d4vd - Fly Away", I have to explicitly search by the name of the song, so "Fly Away", but if I for example can't remember the song's name but I can remember the artist's name, well searching for "d4vd" shows no results.

Just as I am writing this I tried it in a web browser and it behaves the same... oh well, do as you see fit.

Thanks!

**Comment by Maxplosion** [comment]
Maybe I don't fully get it, but why not search in the Artist tab then, if you know the artists name anyway?

**Comment by Chaphasilor** [comment]
> but why not search in the Artist tab then

Sometimes I want to use the artist name to specify a song with a common title. Searching for "love" and "Mothica" would yield just a single song in my library, while just "love" would yield hundreds of songs

**Comment by SnipeXandrej** [comment]
> Maybe I don't fully get it, but why not search in the Artist tab then, if you know the artists name anyway?

There's a slight problem with that.

I have a playlist named "SnipeX's playlist (Spotify)" which is also automatically made into an album by Jellyfin (thanks! /s) which in this case is for some reason automatically named "King Kais Planet".

Trying to search for a song by the Artist will just show me that "Album" (SnipeX's playlist) which has 2100 songs from all kinds of artists... and clicking on that "album" will just show me all 2100 songs... I think you can see the problem here.

https://github.com/jmshrv/finamp/assets/11980981/b68d0f07-2655-4dfa-8b02-a4d1d2e6b6ee

**Comment by Maxplosion** [comment]
so then a solution might be another tab that offers a combined search where you can specify multiple words and get results from all tabs.

because having a large library and searching for a song where you only know parts of the name, but then getting hundrets of artists thrown into the results sounds not so great to me.

**Comment by SnipeXandrej** [comment]
I think it would be great to have a search button (that would act like a filter, like on spotify for example) inside the playlists themselves.

So you can just simply filter all the songs in the playlist by the Artist or the name of the song.

**Comment by SnipeXandrej** [comment]
Basically this would be perfect

https://github.com/jmshrv/finamp/assets/11980981/e84be349-797b-4aa8-9131-87b2b0c58025

**Comment by Chaphasilor** [comment]
Yeah, I think a separate search page that combines all item types (like Spotify's search) is definitely useful, but a local "filter" that keeps the current behavior (only filter by song name in songs tab, by artists name in artist tab, etc.) is also nice. But we can try out if what @Maxplosion said really is a problem or not :)

**Comment by jmshrv** [comment]
I'd like to redo search as part of the redesign, basically opening a new screen to show combined search results. Would make more sense to the user, and greatly clean up the tab view code since we no longer have to handle search queries in it.

**Comment by bsfreq** [comment]
I've been missing this feature as well. Would be nice if the "xxx Tracks" in artist view would work as a link for listing all artist tracks, and also have the search function in that song listing view.. 
![IMG_8724](https://github.com/jmshrv/finamp/assets/163302697/33fafd97-949e-41ea-842a-5884c67cd688)

**Comment by obiwantoby** [comment]
I’m a firm believer in simplicity by default, with an option to keep the legacy experience as is. I haven’t been tracking the redesign closely, but I did submit a PR to the main branch that adds an optional universal search. In my view it’s a better default, while some users will still prefer the existing segmented tabs.

I suggest we accept the PR and iterate on the experience. It’s already good, and over‑analysis will stall progress.

https://github.com/user-attachments/assets/09a2e517-fb35-4338-a2a4-960831953a60
